services/appflowy-collaborate/src/metrics.rs

```diff
@@ -112,8 +122,22 @@ impl CollabRealtimeMetrics {
       "number of collab loads (with history)",
       metrics.load_full_collab_count.clone(),
     );
+    realtime_registry.register(
+      "collab_stream_latency",
+      "latency since collab update is send to a stream to be read from it",
+      metrics.collab_stream_latency.clone(),
+    );
     metrics
   }
+
+  pub fn observe_collab_stream_latency(&self, message_id_timestamp: u64) {
+    let now = Utc::now().timestamp_millis() as u64;
+    if now > message_id_timestamp {
```

Comment:
Under what circumstances will the message timestamp to be greater than the current time?

Reply:
If clocks between Redis and appflowy-cloud processes are desynchronided, NTP clock correction happens for any reason or if the leaps seconds happens.